app/time_tracking/work_statistic/views.py

```diff
@@ -0,0 +1,148 @@
+            end_time__gt=timezone.now() - timedelta(days=356)
+        ).aggregate(Sum('work_time'))
+        responce = {
+            "total_working_time_in_seconds": work_times['work_time__sum']
+        }
+
+        return Response(responce, status.HTTP_200_OK)
+
+    return Response(status=status.HTTP_400_BAD_REQUEST)
+
+
+
+@api_view(['GET'])
+@permission_classes((IsAuthenticated, ))
+def employees_arrival_and_leving_time_statistics_detail(request, format=None):
+    """
+    Retrieve the employees arrival time and leaving times.
+    """
+    if request.method == 'GET':
+        responce = []
+        for user in User.objects.filter(is_staff=False):
+            avg_work_time_starts = WorkTime.objects.filter(owner=user).aggregate(Avg('start_unix_time'))['start_unix_time__avg']
+            avg_work_time_ends = WorkTime.objects.filter(owner=user).aggregate(Avg('end_unix_time'))['end_unix_time__avg']
+
+            if avg_work_time_starts is not None:
+                avg_work_time_starts = datetime.fromtimestamp(avg_work_time_starts)
+            if avg_work_time_ends is not None:
+                avg_work_time_ends = datetime.fromtimestamp(avg_work_time_ends)
+
+            responce.append({
+                'user': user.username,
+                'avarage_arraival_time': avg_work_time_starts,
+                'avarage_leaving_time': avg_work_time_ends,
+            })
+
+        return Response(responce, status.HTTP_200_OK)
+
+    return Response(status=status.HTTP_400_BAD_REQUEST)
+
+
+
+@api_view(['GET'])
+@permission_classes((IsAuthenticated, ))
+def working_ours_to_leaving_hours_detail(request, format=None):
+    """
+    Retrieve the team working hours to leaving hours
+    """
+    if request.method == 'GET':
+        working_hours = 0
+        leaving_hours = 0
+        for user in User.objects.filter():
+            work_times_grouped_by_owner_and_date = WorkTime.objects.filter(
+                owner=user,
+                end_unix_time__isnull=False,
+                start_unix_time__isnull=False
+            ).values(
+                'days_count', 'start_unix_time', 'end_unix_time'
+            ).annotate(days=Count('days_count')).order_by('-start_unix_time')
+
+            length = len(work_times_grouped_by_owner_and_date)
+            index = 0
+
+            for work_time in work_times_grouped_by_owner_and_date:
+                working_hours += work_time['end_unix_time'] - work_time['start_unix_time']
+                if ++index < length - 1:
+                    leaving_hours += work_times_grouped_by_owner_and_date[index]['start_unix_time'] - work_time['end_unix_time']
+
+        if working_hours is 0 or leaving_hours is 0:
+            work_over_leav = 0
+        else:
+            work_over_leav = working_hours / leaving_hours
+        responce = {
+            'working_time': working_hours,
+            'leaving_time': leaving_hours,
+            'work_on_leav_time': work_over_leav,
```

Comment:
should be written completely `work_on_leave_time`

Reply:
Sorry, I'm not sure I understood, could you please elaborate on this?

Reply:
I was referring to `leav` (could be a typo)

Reply:
Ooh, yes it was a typo, and I fixed it on the `develop` branch.

Reply:
The parameters have been renamed to
```python
        response = {
            'working_hours': stats['work_hours'],
            'leaving_hours': stats['leave_hours'],
            'work_on_leave_hours': work_over_leave,
        }
```